examples/scene.rs

```diff
@@ -0,0 +1,187 @@
+use common::*;
+
+fn main() {
+    App::new()
+        .init_resource::<Counter>()
+        .add_plugins((
+            DefaultPlugins.set(ImagePlugin::default_nearest()),
+            QuillPlugin,
+        ))
+        // .add_plugins((CorePlugin, InputPlugin, InteractionPlugin, BevyUiBackend))
+        .add_systems(Startup, (setup, setup_view_root))
+        .add_systems(Update, (close_on_esc, rotate, update_counter))
+        .run();
+}
+
+fn setup_view_root(mut commands: Commands) {
+    commands.spawn(
+        Element::<SpatialBundle>::new()
+            .insert(Name::from("view_root"))
+            .children(RootWidget)
+            .to_root(),
+    );
+
+    commands.spawn(
+        Element::<NodeBundle>::new()
+            .children(InstructionRoot)
+            .to_root(),
+    );
+}
+
+// Counter containing global state.
+#[derive(Resource, Default, PartialEq)]
+pub struct Counter {
+    pub count: usize,
+}
+
+fn update_counter(mut counter: ResMut<Counter>, key: Res<ButtonInput<KeyCode>>) {
+    if key.just_pressed(KeyCode::Space) {
+        counter.count += 1;
+    }
+}
+
+#[derive(Clone, PartialEq)]
+struct RootWidget;
+
+impl ViewTemplate for RootWidget {
+    type View = impl View;
+
+    fn create(&self, cx: &mut Cx) -> Self::View {
+        let counter = cx.use_resource::<Counter>();
+        let count = counter.count;
+
+        const STEP_WIDTH: f32 = 1.5;
+        let x_offset = (STEP_WIDTH * count as f32) / 2.;
+
+        Element::<SpatialBundle>::new().children((
+            // Always show meshcube widget
+            MeshCube {
+                position: Vec3::new(-1.5, 3., 2.),
+            },
+            // Conditionally show MeshCube widget.
+            Cond::new(
+                count % 2 == 0,
+                MeshCube {
+                    position: Vec3::new(1.5, 3., 2.),
+                },
+                (),
+            ),
+            // Keeps children centered
+            Element::<SpatialBundle>::new()
+                .insert_dyn(
+                    move |x_offset| Transform {
+                        translation: Vec3::new(-x_offset, 0., 2.),
+                        ..Default::default()
+                    },
+                    x_offset,
+                )
+                .children(
+                    // Loop over data, (in this case a simple range) and show
+                    // a MeshCube for each element.
+                    For::each(0..count, move |i| MeshCube {
+                        position: Vec3::new(STEP_WIDTH * *i as f32, 1., 0.),
+                    }),
+                ),
+        ))
+    }
+}
+
+#[derive(Clone, PartialEq, Debug)]
+struct MeshCube {
+    position: Vec3,
+}
+
+impl ViewTemplate for MeshCube {
+    type View = impl View;
+    fn create(&self, cx: &mut Cx) -> Self::View {
+        let scale = cx.create_mutable(Vec3::new(1., 1., 1.));
+
+        // let handle_out = cx.create_callback(move || {
```

Comment:
The problem is that the closure is trying to capture `cx`. The solution is to inject `World` and then use `world` instead of `cx`.